device_doctor/lib/src/ios_device.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show LineSplitter;
+
+import 'package:meta/meta.dart';
+
+import 'device.dart';
+import 'process_helper.dart';
+
+/// IOS implementation of [DeviceDiscovery].
+///
+/// Discovers available ios devices and chooses one to work with.
+class IosDeviceDiscovery implements DeviceDiscovery {
+  factory IosDeviceDiscovery() {
```

Comment:
What is the goal of having this be a singleton?

Reply:
This is mirrored over from existing cocoon/agent. My guess is to ensure only one `IosDeviceDiscovery` instance. @yjbanov may have a better idea?